src/core/Akka.TestKit/Internal/InternalTestActor.cs

```diff
@@ -56,16 +56,26 @@ protected override bool Receive(object message)
             {
                 case TestActor.SetIgnore setIgnore:
                     _ignore = setIgnore.Ignore;
-                    return true;
+                    return;
                 case TestActor.Watch watch:
                     Context.Watch(watch.Actor);
-                    return true;
+                    RunTask(async () =>
```

Comment:
Asynchronously run a messaging loop back to the actor being watched in order to make sure that our `Watch` has, in fact, gone through.

Reply:
Loop may not even be necessary - just completing the `TCS` might be good enough.

Reply:
Fixed this and simplified